### PR TITLE
fix(treesitter): update parser and queries

### PR DESCRIPTION
**Pull request by clason**
- build(deps): update lua parser and queries
- fix(treesitter): update vimdoc and vimscript queries
- fix(treesitter): update c queries

### CODE REVIEW

**Comment by clason** [comment]
@bfredl @gpanders @lewis6991

**Review by lewis6991 — APPROVED** [review]

File: runtime/lua/vim/treesitter/query.lua [hunk 1]
```diff
@@ -382,6 +382,36 @@ local predicate_handlers = {
...
+  ['has-ancestor?'] = function(match, _, _, predicate)
+    local node = match[predicate[2]]
+    if not node then
+      return true
+    end
+
+    local ancestor_types = { unpack(predicate, 3) }
+    node = node:parent()
+    while node do
+      if vim.tbl_contains(ancestor_types, node:type()) then
```
Comment (vigoux) [inline runtime/lua/vim/treesitter/query.lua:395]: Just passing by, but can't this be made a lot faster by turning the `ancestor_type` into a map ?
Something along those line:
```lua
for _, ty in ipairs(ancestor_types) do
  map[ty] = true
end
```
And then use it directly here:
```suggestion
      if map[node:type()] then
```

File: runtime/lua/vim/treesitter/query.lua [hunk 2]
```diff
@@ -382,6 +382,36 @@ local predicate_handlers = {
...
+  end,
+
+  ['has-parent?'] = function(match, _, _, predicate)
+    local node = match[predicate[2]]
+    if not node then
+      return true
+    end
+
+    local ancestor_types = { unpack(predicate, 3) }
+    if vim.tbl_contains(ancestor_types, node:parent():type()) then
```
Comment (vigoux) [inline runtime/lua/vim/treesitter/query.lua:410]: Same remark as above.

**Comment by clason** [comment]
Note to self: new predicates also need documentation

File: runtime/lua/vim/treesitter/query.lua (hunk 1 above)
Comment (clason) [reply runtime/lua/vim/treesitter/query.lua:395]: Depends on how large the list is, possibly. (In any case, this _is_ a list, so `list_contains` should be used.)

File: runtime/lua/vim/treesitter/query.lua (hunk 2 above)
Comment (clason) [reply runtime/lua/vim/treesitter/query.lua:410]: Not sure it's worth it here, as we're only doing this once (should be `list_contains`, though)?

**Pull request merged**